### PR TITLE
Add exact and strict support to renderRoutes

### PR DESCRIPTION
**Pull request by lith-light-g**
My attempt at writing tests for #4728

### CODE REVIEW

File: packages/react-router-config/modules/__tests__/integration-test.js [hunk 1]
```diff
@@ -99,5 +99,107 @@ describe('integration', () => {
...
+        component: Comp,
+        exact: true,
+        routes: [
+          {
+            path: '/pepper/child',
+            component: Comp
+          }
+        ]
+      },
+      {
```
Comment (pshrmn) [inline packages/react-router-config/modules/__tests__/integration-test.js]: I don't think that this second route is necessary.

**Comment by lith-light-g** [comment]
I wrote tests for `renderRoutes` in a separate file. And changed the integration tests I wrote earlier to test `exact` + `strict`.

File: packages/react-router-config/modules/__tests__/renderRoutes-test.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import { renderToString } from 'react-dom/server'
+import StaticRouter from 'react-router/StaticRouter'
+
+describe('renderRoutes', () => {
+  let rendered
+  const Comp = ({ match, route, route: { routes } }) => (
+    rendered.push(route),
+    renderRoutes(routes)
+  )
+  beforeEach(() => rendered = [])
```
Comment (pshrmn) [inline packages/react-router-config/modules/__tests__/renderRoutes-test.js]: Every other before/after each call uses this style.
```js
beforeEach(() => {
  rendered = []
})
```

File: packages/react-router-config/modules/__tests__/renderRoutes-test.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    it('renders the first matched route', () => {
+      const routeToMatch = {
+        component: Comp,
+        path: '/'
+      }
+      const routes = [routeToMatch, {
+        component: Comp,
+      }]
+
+      renderToString(<StaticRouter location='/' context={{}}>{renderRoutes(routes)}</StaticRouter>)
```
Comment (pshrmn) [inline packages/react-router-config/modules/__tests__/renderRoutes-test.js]: No need to cram everything into one line.

```js
renderToString(
  <StaticRouter location='/' context={{}}>
    {renderRoutes(routes)}
  </StaticRouter>
)
```

File: packages/react-router-config/modules/__tests__/renderRoutes-test.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    const routeToMatch = {
+      component: Comp
+    }
+    const routes = [routeToMatch]
+
+    renderToString(<StaticRouter location='/path' context={{}}>{renderRoutes(routes)}</StaticRouter>)
+    expect(rendered.length).toEqual(1)
+    expect(rendered[0]).toMatch(routeToMatch)
+  })
+  describe('Switch usage', () => {
```
Comment (pshrmn) [inline packages/react-router-config/modules/__tests__/renderRoutes-test.js]: Leave a blank line before the `describe`.

File: packages/react-router-config/modules/__tests__/renderRoutes-test.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      }
+      const routes = [routeToMatch, {
+        component: Comp,
+      }]
+
+      renderToString(<StaticRouter location='/' context={{}}>{renderRoutes(routes)}</StaticRouter>)
+      expect(rendered.length).toEqual(1)
+      expect(rendered[0]).toMatch(routeToMatch)
+    })
+    it('renders the first matched route in nested routes', () => {
```
Comment (pshrmn) [inline packages/react-router-config/modules/__tests__/renderRoutes-test.js]: Blank line here, too.

**Comment by lith-light-g** [comment]
Should be good to go !

**Comment by timdorr** [comment]
That's a lot of tests! LGTM too. Good work, @lith-light-g!

**Comment by amurdock** [comment]
Any idea when this PR will find its way into a release ... would really help me out with a server rendering example I'm writing?

**Comment by lennerd** [comment]
+1 for more info when this will be released.

**Comment by amurdock** [comment]
If anyone is interested I've forked and published as @amurdock/react-router-config until the next release.

**Pull request merged**